src/lib/routeLoaders.ts

```diff
@@ -44,34 +62,33 @@ export const onboardingRedirectLoader: ActionFunction = async ({ request }) => {
     )
   }
 
-  return null
+  return settingsLoader(args)
 }
 
 export const fileLoader: LoaderFunction = async ({
   params,
-}): Promise<IndexLoaderData | Response> => {
-  const { settings } = await loadAndValidateSettings()
+}): Promise<FileLoaderData | Response> => {
+  let settings = await loadAndValidateSettings()
 
-  const defaultDir = settings.defaultDirectory || ''
+  const defaultDir = settings.app.projectDirectory.current || '/'
+  const projectPathData = getProjectMetaByRouteId(params.id, defaultDir)
+  const isBrowserProject = params.id === decodeURIComponent(BROWSER_PATH)
 
-  if (params.id && params.id !== BROWSER_FILE_NAME) {
-    const decodedId = decodeURIComponent(params.id)
-    const projectAndFile = decodedId.replace(defaultDir + sep, '')
-    const firstSlashIndex = projectAndFile.indexOf(sep)
-    const projectName = projectAndFile.slice(0, firstSlashIndex)
-    const projectPath = defaultDir + sep + projectName
-    const currentFileName = projectAndFile.slice(firstSlashIndex + 1)
+  if (!isBrowserProject && projectPathData) {
+    const { projectName, projectPath, currentFileName, currentFilePath } =
+      projectPathData
 
-    if (firstSlashIndex === -1 || !currentFileName)
+    if (!currentFileName || !currentFilePath) {
       return redirect(
         `${paths.FILE}/${encodeURIComponent(
-          `${params.id}${sep}${PROJECT_ENTRYPOINT}`
+          `${params.id}${isTauri() ? sep : '/'}${PROJECT_ENTRYPOINT}`
         )}`
       )
+    }
 
     // TODO: PROJECT_ENTRYPOINT is hardcoded
```

Comment:
Actually wherever that other one was probably should stay for the time being. I had planned on including a `projectEntrypoint` setting so that users could say that this project's default file is, say, `rocket.kcl` instead of `main.kcl`.

But it actually has a ton more edge cases and I think I should have a "no file open" but "project open" state designed and implemented before I turn that setting on, because without it it would be very possible to get your project into a broken state that you need to edit the `project.toml` to bail out of.

Reply:
Just went back to double-check and this spot is still okay to remove because it was hardcoding the LSP's file handle to be `PROJECT_ENTRYPOINT` until we supported multiple files, which we do now so it's since been made to be based on the current URL.